### DISCUSSION TITLE
[Buildings] Coverage maps

### DISCUSSION THREAD

**Discussion by calebrob6** [OvertureMaps/data#250] [Discussion]
### Category Feedback

Is there a way to estimate whether or not a given location has Building coverage? Alternatively, is there a list of areas that are known to have incomplete or poor coverage? (e.g. Valencia, Spain or most of China)

**Comment by jenningsanderson** [comment]
Great question, @calebrob6. 

Determining building coverage or completeness is a complex task. A proper analysis should also compare building counts to population density as well, that said — if you want a general idea of Overture building coverage, here's a script you can run in DuckDB that can produce a basic coverage map: 

```sql
INSTALL h3 FROM community;
LOAD h3;
COPY(
  SELECT
    h3_latlng_to_cell_string(bbox.ymin, bbox.xmin, 7) as h3,
    count(1) AS buliding_count
  FROM read_parquet('s3://overturemaps-us-west-2/release/2024-11-13.0/theme=buildings/type=building/*')
  WHERE bbox.xmin BETWEEN -10.18 AND 3.48
    AND bbox.ymin BETWEEN 35.89 AND 43.99
  GROUP BY 1
) TO 'spain_buildings_h3_density.csv';
```

The resulting file can be loaded directly into kepler.gl for a quick building density map of Spain: 
![image](https://github.com/user-attachments/assets/21d5759b-d49c-466a-a835-9fb11634b726)
(Map data from OpenStreetMap via Overture Maps)

**Reply by calebrob6** [reply]
Thanks @jenningsanderson -- I was getting at what you are calling "completeness" I think (cool visualization by the way 😄). I was imagining the current coverage map diffed with coarse building density (maybe a dataset like GHSL https://human-settlement.emergency.copernicus.eu/).
